src/router/graphql.ts

```diff
@@ -0,0 +1,33 @@
+import * as express from 'express'
+import { postgraphile, } from 'postgraphile'
+
+import * as dbMan from '../services/db-manager';
+import config from '../config'
+import metadata from '../services/repositories/metadata'
+
+export const graphql = express.Router();
+
+if (config.graphql) {
+  const withGraphiql = config.graphql.withGraphiql;
+  // TODO: Turn this into an expiring map when activity is low
+  const postgraphiles: { [key: string]: any, } = {};
+
+  const getPostgraphile = (db: string) => {
+    postgraphiles[db] = postgraphiles[db] ?? postgraphile(
+      `postgresql://${config.db.user}:${config.db.password}@${config.db.host}/${db}${config.db.forceSSL ? '?sslmode=require' : ''}`,
+      'public',
+      {
+        externalUrlBase: `/v1/graphql/${db}`,
+        graphiql: withGraphiql,
+      });
+    return postgraphiles[db];
+  };
+
+  graphql.use('/:db', async (req, res, next) => {
+    const db = config.auth0 ?
+      // This will throw an error if the user is not allowed to access the specified DB
```

Comment:
there is no try catch, does the error crash the process?

Reply:
I assumed this would handle it? https://github.com/iasql/iasql-engine/blob/main/src/index.ts#L43-L52

It's a thrown error from an Express router and I thought Express grabbed the exceptions for us to route into that thing.

Reply:
Got it, I think we should be fine then!